### PR TITLE
Verify spec

### PR DESCRIPTION
**Pull request by FynnBe**
also contains:
- changes to adhere to format version 0.3.0, which is not yet finalized.
- draft of a runner for scikit-learn models (we are not planning to officially support sklearn models at this point)

### CODE REVIEW

**Comment by oeway** [comment]
Nice, we should also add this verification to the compilation script and generate a report for each model (e.g. highlight as an error icon on the model card if error detected).

**Comment by oeway** [comment]
@FynnBe Can you generate an error report after verification, not only throw error but output details about the error?

**Comment by FynnBe** [comment]
> @FynnBe Can you generate an error report after verification, not only throw error but output details about the error?

example from travis:
image

**Comment by constantinpape** [comment]
The checker works for my most current v3 spec:
https://github.com/hci-unihd/batchlib/blob/master/misc/models/torch/bioimageio/UNetCovidIf.model.yaml

What's left to do before merging?

**Comment by FynnBe** [comment]
> What's left to do before merging?

get the tests to pass...

File: pybio/core/datasets/base.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import collections
+from typing import Dict, Optional, OrderedDict, Sequence, Tuple
```
Comment (m-novikov) [inline pybio/core/datasets/base.py:2]: Use Dict instead of OrderedDict

Comment (FynnBe) [reply pybio/core/datasets/base.py:2]: I went with MutableMapping instead, but seem to have forgotten it here... I'll fix it

Comment (m-novikov) [reply pybio/core/datasets/base.py]: I don't know if MutableMapping implies ordering, while python Dict does.

**Comment by FynnBe** [comment]
I am a bit puzzled why this fails:
```
self = PosixPath('/home/travis/build/bioimage-io/python-bioimage-io/tests/conftest.py/../../specs/models/sklearn/RandomForestClassifier_v0_1.model.yaml')
name = '/home/travis/build/bioimage-io/python-bioimage-io/tests/conftest.py/../../specs/models/sklearn/RandomForestClassifier_v0_1.model.yaml'
flags = 524288, mode = 438
    def _opener(self, name, flags, mode=0o666):
        # A stub for the opener argument to built-in open()
>       return self._accessor.open(self, flags, mode)
E       NotADirectoryError: [Errno 20] Not a directory: '/home/travis/build/bioimage-io/python-bioimage-io/tests/conftest.py/../../specs/models/sklearn/RandomForestClassifier_v0_1.model.yaml'
```

**Comment by FynnBe** [comment]
> I am a bit puzzled why this fails:

for PosixPaths this does not seem legal: `posix_path/a_file.ext/../elsewhere`

**Pull request merged**